tests/nlu/selectors/test_selectors.py

```diff
@@ -268,105 +294,176 @@ async def test_train_model_checkpointing(
-    component_builder: ComponentBuilder,
-    tmp_path: Path,
+    config_params: Dict[Text, Any],
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
     should_finetune: bool,
 ):
-    _config = RasaNLUModelConfig({"pipeline": pipeline, "language": "en"})
+    loaded_pipeline = [
+        registry.get_component_class(component.pop("name"))(component)
+        for component in copy.deepcopy(pipeline)
+    ]
 
-    (trainer, trained, persisted_path) = await rasa.nlu.train.train(
-        _config,
-        path=str(tmp_path),
-        data="data/examples/rasa/demo-rasa.yml",
-        component_builder=component_builder,
+    importer = RasaFileImporter(
+        training_data_paths=["data/examples/rasa/demo-rasa.yml"]
+    )
+    training_data = importer.get_nlu_data()
+
+    for component in loaded_pipeline:
+        component.train(training_data)
+
+    if should_finetune:
+        default_execution_context.is_finetuning = True
+
+    response_selector = create_response_selector(
+        config_params, default_model_storage, default_execution_context
     )
+    response_selector.train(training_data=training_data)
 
-    assert trainer.pipeline
-    assert trained.pipeline
+    message = Message(data={TEXT: "Rasa is great!"})
 
-    loaded = Interpreter.load(
-        persisted_path,
-        component_builder,
-        new_config=_config if should_finetune else None,
+    for component in loaded_pipeline:
+        component.process(message)
+
+    message2 = copy.deepcopy(message)
+
+    classified_message = response_selector.process([message])[0]
+
+    loaded_selector = ResponseSelectorGraphComponent.load(
+        {**ResponseSelectorGraphComponent.get_default_config(), **config_params},
+        default_model_storage,
+        Resource("response_selector"),
+        default_execution_context,
     )
 
-    assert loaded.pipeline
-    assert loaded.parse("Rasa is great!") == trained.parse("Rasa is great!")
+    classified_message2 = loaded_selector.process([message2])[0]
+
+    assert classified_message2.fingerprint() == classified_message.fingerprint()
 
 
 @pytest.mark.skip_on_windows
-async def test_train_persist_load(component_builder: ComponentBuilder, tmpdir: Path):
+def test_train_persist_load(
+    default_model_storage: ModelStorage, default_execution_context: ExecutionContext,
+):
     pipeline = [
         {"name": "WhitespaceTokenizer"},
         {"name": "CountVectorsFeaturizer"},
-        {"name": "ResponseSelector", EPOCHS: 1},
     ]
-    await _train_persist_load_with_different_settings(
-        pipeline, component_builder, tmpdir, False
-    )
-    await _train_persist_load_with_different_settings(
-        pipeline, component_builder, tmpdir, True
+    config_params = {EPOCHS: 1}
+
+    _train_persist_load_with_different_settings(
+        pipeline, config_params, default_model_storage, default_execution_context, False
     )
 
+    # _train_persist_load_with_different_settings(
```

Comment:
This part of the test fails with `InvalidParameterException` raised by `DietClassifierGraphComponent` in L314  - unsure how to adapt this further 🙈